src/main/java/frc/robot/subsystems/Auto.java

```diff
@@ -14,32 +14,37 @@
     private double leftTarget;
     private double rightTarget;
 
-    static final int WAIT = 0;
-    static final int DRIVE = 1;
-    static final int TURN = 2;
-    static final int START_GATHERER = 3;
-    static final int STOP_GATHERER = 4;
-    static final int START_FLYWHEEL = 5;
-    static final int STOP_FLYWHEEL = 6;
-    static final int SHOOT = 7;
-    static final int DRIVE_BALL = 8;
-    static final int DRIVE_HOOP = 9;
-
-    static final int FEEDER_CYCLES = 75;
-
-    static final int MAX_TURN_SECONDS = 3;
-    static final int MAX_BALL_SECONDS = 5;
+    private static final int WAIT = 0;
+    private static final int DRIVE = 1;
+    private static final int TURN = 2;
+    private static final int START_GATHERER = 3;
+    private static final int STOP_GATHERER = 4;
+    private static final int START_FLYWHEEL = 5;
+    private static final int STOP_FLYWHEEL = 6;
+    private static final int SHOOT = 7;
+    private static final int DRIVE_BALL = 8;
+    private static final int DRIVE_HOOP = 9;
+
+    private static final int FEEDER_CYCLES = 75;
+
+    private static final int MAX_TURN_SECONDS = 3;
+    private static final int MAX_BALL_SECONDS = 5;
 
     private OperatorInterface oi;
     private Shooter shooter;
     private Chassis chassis;
 
     private VisionData vData;
 
-    private int[][] steps;
+    private double[][] steps;
+
+    // No Auto
+    public static final double[][] noAuto = {
+    };
+
     // Start gatherer, drive X feet, stop gatherer, start flywheel at known RPM,
     // turn 180, shoot, stop flywheel
-    public static int[][] basicAutoSteps = {
+    public static final double[][] basicAutoSteps = {
```

Comment:
These should stay as ints, nothing here needs the precision, and we are using enumerated constants and it is a little weird to store them as doubles.

Reply:
it threw errors when it wasn't a double in the constructors

Reply:
I rounded the two turns to the nearest degree and the compiler seems happy now.

Reply:
cool